feat(about-pages): add custom about pages

TheChilledBuffalo opened the pull request.

dominichayesferen commented:
Shouldn't this also be used to override about:newtab and about:home?

TheChilledBuffalo commented:
I did speak to the dev about that. But since the main purpose of this is to hide the unprofessional looking URL and since the URL of Floorp Start (the new tab/home) is hidden by default, apparently it's not necessary. Only the upcoming Floorp Welcome feature is currently planned to be added here. Open to suggestions.

dominichayesferen commented:
> I did speak to the dev about that. But since the main purpose of this is to hide the unprofessional looking URL and since the URL of Floorp Start (the new tab/home) is hidden by default, apparently it's not necessary. Only the upcoming Floorp Welcome feature is currently planned to be added here. Open to suggestions.

@surapunoyousei As pointed out in #70, replacing about:home and about:newtab directly with Floorp Start would instantly fix several issues in one go, as well as fixing the fact there's unintentionally 2 new tab pages currently in Floorp 12 (Floorp's and Firefox's) and cleaning up the new tab page-related defaults by removing hacky defaults used to accommodate for NOT replacing about:home and about:newtab with Floorp Start.

CutterKnife commented:
> fixing the fact there's unintentionally 2 new tab pages currently in Floorp 12 (Floorp's and Firefox's)

I was told that the reason there is Floorp Home and Floorp Start is to take into account those who want Floorp Home.

TheChilledBuffalo commented:
Now that I think about it, @dominichayesferen is right. Floorp should have an identity of its own. I think it would be better to replace Firefox's  new tab entirely with Floorp's.

CutterKnife commented:
Hmm, that's one way to think of it, I figured some people might want something that is unique to Firefox Home.
Recent Activity, weather, and fully customizable shortcuts are only in Home for now (though they may be implemented later).

TheChilledBuffalo commented:
Hmm... that makes sense too. We'll wait for the main dev's opinion.

@surapunoyousei thoughts?

TheChilledBuffalo commented:
Yeah I guess, that would work. We'll wait for Asano-san's opinion, and then proceed further.

surapunoyousei commented:
Hi. Thank you for your discussion!

IMO, replace Firefox Home is better. But, we need to add option "Revert to Floorp Home" to design preferences.

TheChilledBuffalo commented:
I have implemented the overriding of `about:newtab` and `about:home`. I have added a preference in Floorp Hub for it too. I also removed some code which I believe is redundant with the new logic for newtab overriding, though I may be wrong.

This PR requires thorough review before merging.

surapunoyousei commented:
Thanks! I picked all commits!
